### PR TITLE
WIP: Replicating current project as a react application

### PR DESCRIPTION
**Pull request by akshay-nm**
This is related to #1 

# Todo
- [x] Setup react app
- [x] Integrate react router
- [ ] Add routes for all pages
- [ ] Add components as per content

### CODE REVIEW

File: bcnchess-react/src/App.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import './App.css';
+import { BrowserRouter as Router, Route, Redirect, Switch } from 'react-router-dom'
+import Header from './components/Header';
+import Home from './components/home/Home';
+import About from './components/about/About';
+
+function App() {
+  return (
+    <Router>
```
Comment (akshay-nm) [inline bcnchess-react/src/App.js:16]: React router integration

File: bcnchess-react/src/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (akshay-nm) [inline bcnchess-react/src/components/Header.js:1]: Header component

File: bcnchess-react/src/components/about/About.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (akshay-nm) [inline bcnchess-react/src/components/about/About.js]: About page component

File: bcnchess-react/src/components/home/Home.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (akshay-nm) [inline bcnchess-react/src/components/home/Home.js]: Home page component

File: bcnchess-react/src/index.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+import store from './app/store';
+import { Provider } from 'react-redux';
+import * as serviceWorker from './serviceWorker';
+import 'bootstrap/dist/css/bootstrap.min.css';
```
Comment (akshay-nm) [inline bcnchess-react/src/index.js:8]: importing bootstrap

**Comment by akshay-nm** [comment]
This PR is going to be huge. Lemme know if you need help reviewing it.

**Comment by ctfrancia** [comment]
Once it's all done then I'll check out the PR and go through it and make sure that it is working as it should. I am not sure if you could do it but include some Unit Tests with it. That way, we can integrate it with the CI/CD once I get that setup

**Comment by ctfrancia** [comment]
Also I have started the process of including the API documentation with postman. So soon you'll have all the routes, as of now, set up to make requests.

**Comment by akshay-nm** [comment]
We already have redux for client-side state management. I usually put the fetch logic in:
- in a component
- in a redux slice

So integrating with any API shouldn't be a problem as long as we can send HTTP requests to access it.

**Comment by ctfrancia** [comment]
Great. I'll take a look at it once I have some spare time today.

**Comment by ctfrancia** [comment]
Hey so I have started looking at it now. The first thing that I've noticed is that you need to be logged in to see the tournaments. (App.js) What I would like is that they can see the homepage (where they see the list of tournaments) and the option to log in (if they need to).
protected paths/need to be logged in = Create Tournament, user profile

**Comment by akshay-nm** [comment]
okay, lemme have a look at it.

**Comment by akshay-nm** [comment]
Apologies @ctfrancia 
Lost track of this one...

**Pull request closed without merging**